plenum/common/ledger.py

```diff
@@ -0,0 +1,97 @@
+from copy import copy
+from typing import List, Tuple
+
+import base58
+
+from ledger.stores.chunked_file_store import ChunkedFileStore
+from ledger.stores.file_store import FileStore
+
+from ledger.ledger import Ledger as _Ledger
+
+
+class Ledger(_Ledger):
+    @staticmethod
+    def _defaultStore(dataDir, logName, ensureDurability) -> FileStore:
+        return ChunkedFileStore(dataDir,
+                                logName,
+                                isLineNoKey=True,
+                                storeContentHash=False,
+                                ensureDurability=ensureDurability)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Merkle tree of containing transactions that have not yet been
+        # committed but optimistically applied.
+        self.uncommittedTxns = []
+        self.uncommittedRootHash = None
+        self.uncommittedTree = None
+
+    @property
+    def uncommittedSize(self):
+        return len(self.uncommittedTxns)
+
+    def appendTxns(self, txns: List):
+        # These transactions are not yet committed so they do not go to
+        # the ledger
+        uncommittedSize = self.size + len(self.uncommittedTxns)
+        self.uncommittedTree = self.treeWithAppliedTxns(txns,
+                                                        self.uncommittedTree)
+        self.uncommittedRootHash = self.uncommittedTree.root_hash
+        self.uncommittedTxns.extend(txns)
+        if txns:
+            return (uncommittedSize+1, uncommittedSize+len(txns)), txns
+        else:
+            return (uncommittedSize, uncommittedSize), txns
+
+    def commitTxns(self, count: int) -> Tuple[Tuple[int, int], List]:
+        """
+        The number of txns from the beginning of `uncommittedTxns` to commit
+        :param count:
+        :return: a tuple of 2 seqNos indicating the start and end of sequence
+        numbers of the committed txns
+        """
+        committedSize = self.size
+        for txn in self.uncommittedTxns[:count]:
+            self.append(txn)
+        committedTxns = self.uncommittedTxns[:count]
+        self.uncommittedTxns = self.uncommittedTxns[count:]
+        if not self.uncommittedTxns:
```

Comment:
Should we update `self.uncommittedTree` and `self.uncommittedRootHash` if `self.uncommittedTxns` is still not empty?